From 30b6a5f7948337e12e14fbfd7399f6d6a04bf265 Mon Sep 17 00:00:00 2001
From: Nikolay Borisov <nborisov@suse.com>
Date: Wed, 29 Mar 2017 16:18:31 +0300
Subject: [PATCH] kabi: Protect xfs_mount and xfs_buftarg
References: bsc#1024508
Patch-mainline: Never, kabi

Upstream commit 78c931b8be75 ("xfs: replace global xfslogd wq with
per-mount wq") added another workqueue into struct xfs_mount. Commit
9c7504aa72b6 ("xfs: track and serialize in-flight async buffers against
unmount") in turn added a percpu counter in xfs_buftarg. Since those are
private xfs struct protect them by ifdef guards.

Signed-off-by: Nikolay Borisov <nborisov@suse.com>
---
 fs/xfs/linux-2.6/xfs_buf.h | 2 ++
 fs/xfs/xfs_mount.h         | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/fs/xfs/linux-2.6/xfs_buf.h b/fs/xfs/linux-2.6/xfs_buf.h
index a5edb1bbf460..02b280d8d4a5 100644
--- a/fs/xfs/linux-2.6/xfs_buf.h
+++ b/fs/xfs/linux-2.6/xfs_buf.h
@@ -121,7 +121,9 @@ typedef struct xfs_buftarg {
 	struct list_head	bt_lru;
 	spinlock_t		bt_lru_lock;
 	unsigned int		bt_lru_nr;
+#ifndef __GENKSYMS__
 	struct percpu_counter	bt_io_count;
+#endif
 } xfs_buftarg_t;
 
 struct xfs_buf;
diff --git a/fs/xfs/xfs_mount.h b/fs/xfs/xfs_mount.h
index 52b07a2a1222..1afa2f962c78 100644
--- a/fs/xfs/xfs_mount.h
+++ b/fs/xfs/xfs_mount.h
@@ -215,8 +215,10 @@ typedef struct xfs_mount {
 	int64_t			m_low_space[XFS_LOWSP_MAX];
 						/* low free space thresholds */
 
+#ifndef __GENKSYMS__
 	const char 		*m_buf_workqueue_name;
 	struct workqueue_struct *m_buf_workqueue;
+#endif
 	const char		*m_data_workqueue_name;
 	struct workqueue_struct	*m_data_workqueue;
 	const char		*m_unwritten_workqueue_name;
-- 
2.7.4

